projects/media-viewer/src/lib/media-viewer/media-viewer.model.ts

```diff
@@ -41,3 +42,23 @@ export interface SearchResultsCount {
   current: number;
   total: number;
 }
+
+export class ChangePageOperation {
+  constructor(public changePageParameter: ChangePageParameter) {
```

Comment:
IMO this is a slightly odd inheritance model - it's probably best to have no inheritance and use a union type:

```
public changePageParameter: SetCurrentPage | ChangeByDelta
```

Reply:
alternatively, you could not multi-plex different operations into a single stream